pkg/api/types.go

```diff
@@ -1380,3 +1381,15 @@ const (
 
 	PortHeader = "port"
 )
+
+// ResourceStats represents compute resource usage metrics associated with a pod or a container.
+type ResourceStats struct {
+	// Detailed instantaneous metrics.
+	Detailed *cadvisor_api.ContainerInfo `json:"detailed,omitempty"`
```

Comment:
What this current PR does is wrong in multiple ways -- and so are the current Kubelet APIs. 

For now, I proposed that Vish expose the actual cadvisor API at a new API prefix. The cadvisor API has its own independent versioning. This would effectively make Kubelet a caching proxy for cadvisor.

This is a bigger discussion, which we should have in another forum.